extensions/pkg/webhook/cmd/options.go

```diff
@@ -215,9 +224,13 @@ func (c *AddToManagerConfig) AddToManager(mgr manager.Manager) ([]admissionregis
 	if err != nil {
 		return nil, nil, errors.Wrapf(err, "could not create webhooks")
 	}
-
 	webhookServer := mgr.GetWebhookServer()
 
+	port := webhookServer.Port
+	if c.Server.ServicePort > 0 {
```

Comment:
Shouldn't we check the mode being `service` as well?

Reply:
The port is relevant for both modes, see https://github.com/gardener/gardener/blob/master/extensions/pkg/webhook/registration.go#L122-L127

Reply:
Yes, but which port? If the mode is URL and the URL is not a service URL, this should be the webhook server port, not the service port, correct? E.g., we should still check for `service` or `url-service`?

Reply:
Yes, good point, thanks. I have removed the port usage for the registration when the mode is URL - it doesn't make sense, and introducing additional CLI flags would overcomplicate it, I guess. I'll add a release note that the port must be part of the URL when the URL mode is used, thanks.

Reply:
Hmm, as you wish, though I am not particularly in favor of this. For mode `url`, you would need to pass the port twice - once in the URL itself, and another time as `webhook-server-port` for the port the webhook server to listen on, then you could do a mistake and pass a different port.

Reply:
Also, I think this would then require a change in the `Makefile` files in most of the extensions, see e.g. https://github.com/gardener/gardener-extension-provider-azure/blob/255a61a772cd9ef489129d7794e3f5d656647e47/Makefile#L27-L52

Reply:
Hm, if the default HTTPS port is used (443) which is the case for all extensions then no change is needed, or?

Reply:
But it's not the case, you can see `--webhook-config-server-port=8443` in the above link.

Reply:
OK, yes, true. Well, then that's the price that this must be added. I will take care when I revendor the extensions. The hostname (`docker.for.mac.localhost` should be anyways replaced by `host.docker.internal`, ref #2641). OK?

Reply:
OK, thanks